src/lib.rs

```diff
@@ -109,18 +191,18 @@ pub(crate) struct Outbound {
 }
 
 impl Outbound {
-    #[inline(always)]
+    #[inline]
     fn write_response(&mut self, subj: &str, msgb: &[u8]) -> io::Result<()> {
         write!(self.writer, "PUB {} {}\r\n", subj, msgb.len())?;
-        self.writer.write(msgb)?;
-        self.writer.write(b"\r\n")?;
+        self.writer.write_all(msgb)?;
+        self.writer.write_all(b"\r\n")?;
```

Comment:
This is an actual bug, because `write` may return before completely writing the desired bytes, resulting in short writes. `write_all` loops until the full buffer is written.

Reply:
Even for buffered writers? Also does this cause large perf impact?

Reply:
Yeah, if the written bytes exceed the BufWriter's inner capacity it will skip the inner buffer and go directly to the backing `Write` implementation, which only guarantees that it will return the number of bytes written. 

https://doc.rust-lang.org/src/std/io/buffered.rs.html#615

I would be pretty surprised if the perf impact caused a throughput drop. In the end, it's necessary for correct behavior, but we can avoid `BufWriter` and do all kinds of other fun stuff if that becomes an issue. If we can get that to show up on flamegraphs at all, we're doing a lot of things extremely well :)

Reply:
Gotcha.